### PR TITLE
Improve interactivity while publishing

### PR DESCRIPTION
**Pull request by thejameskyle**
See: https://www.dropbox.com/s/y33j4lh1tikfrxw/lerna-prompts.mov?dl=0

### CODE REVIEW

File: lib/utils/npmUtils.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ var installInDir = logger.logifyAsync("npmUtils.installInDir", function (dir, ca
   });
 });
 
-var addDistTag = logger.logifySync("npmUtils.removeDistTag", function (packageName, version, tag) {
+var addDistTag = logger.logifySync("npmUtils.addDistTag", function (packageName, version, tag) {
```
Comment (hzoo) [inline lib/utils/npmUtils.js]: lol..

Comment (thejameskyle) [inline lib/utils/npmUtils.js]: cher-lloyd-cl-bear-VC0UGtz5v1Is0

File: lib/commands/publish/publishChangedPackages.js [hunk 2]
```diff
@@ -3,33 +3,116 @@ var checkUpdatedPackages  = require("../../utils/checkUpdatedPackages");
...
+    changedPackages.forEach(function (pkg) {
+      versions[pkg.name] = version;
+    });
 
-  var ver = semver.valid(input);
+    callback(null, {
+      version: version,
+      versions: versions
+    });
+  } else if (!flags.independent) {
```
Comment (hzoo) [inline lib/commands/publish/publishChangedPackages.js]: maybe better to switch these and do `flags.independent`?

File: lib/commands/publish/publishChangedPackages.js [hunk 3]
```diff
@@ -60,91 +143,85 @@ function gitRemoveTaggedVersions(tags) {
...
-    if (flags.canary) {
-      // count cross dependent packages as changed packages to be updated as well
-      changedPackages = unionWith(changedPackages, crossDependentUpdates, function(a, b) {
-        return a.loc === b.loc;
+      var version = results.version;
+      var versions = results.versions;
+
+      confirmVersions(changedPackages, versions, function(err, confirmed) {
+        if (err) return onError(err);
+        if (!confirmed) return onSuccess(true);
```
Comment (hzoo) [inline lib/commands/publish/publishChangedPackages.js]: Kinda looks weird to see `onSuccess(true)` when it cancels but not a big deal

**Comment by hzoo** [comment]
👍

File: lib/commands/publish/publishChangedPackages.js (hunk 2 above)
Comment (thejameskyle) [inline lib/commands/publish/publishChangedPackages.js]: Trying to group together canary and non-independent mode since they are similar to one another

**Pull request merged**